packages/html/src/dropdowngrid/dropdowngrid.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import { classNames, States, Size, Roundness, FillMode } from '../utils';
+import {
+    Input,
+    InputPrefix,
+    InputSuffix,
+    InputInnerInput,
+    InputClearValue,
+    InputLoadingIcon,
+    InputValidationIcon
+} from '../input';
+import { Button } from '../button';
+import { Popup } from '../popup';
+
+export const DROPDOWNGRID_CLASSNAME = `k-dropdowngrid`;
+
+const states = [
+    States.hover,
+    States.focus,
+    States.valid,
+    States.invalid,
+    States.required,
+    States.disabled,
+    States.loading
+];
+
+const options = {
+    size: [ Size.small, Size.medium, Size.large ],
+    rounded: [ Roundness.small, Roundness.medium, Roundness.large, Roundness.full ],
+    fillMode: [ FillMode.solid, FillMode.flat, FillMode.outline ],
+};
+
+export type KendoDropdownGridOptions = {
+    size?: (typeof options.size)[number] | null;
+    rounded?: (typeof options.rounded)[number] | null;
+    fillMode?: (typeof options.fillMode)[number] | null;
+};
+
+export type KendoDropdownGridProps = KendoDropdownGridOptions & {
+    prefix?: JSX.Element;
+    suffix?: JSX.Element;
+    type?: string;
+    value?: string;
+    placeholder?: string;
+    autocomplete?: string;
+    popup?: JSX.Element;
+    opened?: boolean;
+};
+
+export type KendoDropdownGridState = { [K in (typeof states)[number]]?: boolean };
+
+export const DropdownGrid = (
+    props: KendoDropdownGridProps &
+        KendoDropdownGridState &
+        React.HTMLAttributes<HTMLSpanElement>
+) => {
+    const {
+        prefix,
+        suffix,
+        value,
+        placeholder,
+        size,
+        rounded,
+        fillMode,
+        hover,
+        focus,
+        valid,
+        invalid,
+        required,
+        loading,
+        disabled,
+        popup,
+        opened,
+        ...other
+    } = props;
+
+
+    return (
+        <>
+            <Input
+                {...other}
+                size={size}
+                rounded={rounded}
+                fillMode={fillMode}
+                hover={hover}
+                focus={focus}
+                valid={valid}
+                invalid={invalid}
+                required={required}
+                loading={loading}
+                disabled={disabled}
+                className={classNames(
+                    props.className,
+                    DROPDOWNGRID_CLASSNAME,
+                    'k-combobox'
```

Comment:
Do we intent to use an `input` with `k-combobx` instead of the `Combobox` spec itself? 

While this might not have an effect on the actual rendering of the component, It would imply the decision when developing the component.

Reply:
IMO, the `combobox` component includes both the input part as well as the list inside popup part. The `dropdowngrid` on the other hand, consists of an input + grid inside popup. This is why I haven't used the `combobox` spec, which is the same only for the input part. We are using the `k-combobox` class for styling purposes. Does it make sense or not really?

Reply:
With the current capabilities of the suits it makes perfect sense. I do believe there is an opportunity there to further improve the "ComboBox" in order to facilitate replacing the content inside the popup - e.g. with a Grid and achieve the DropDownGrid rendering.